test/test-functions

```diff
@@ -633,43 +684,44 @@ install_valgrind() {
         exit 1
     fi
 
-    local _valgrind_bins=$(strace -e execve valgrind /bin/true 2>&1 >/dev/null | perl -lne 'print $1 if /^execve\("([^"]+)"/')
-    dracut_install $_valgrind_bins
+    local valgrind_bins valgrind_libs valgrind_dbg_and_supp
 
-    local _valgrind_libs=$(LD_DEBUG=files valgrind /bin/true 2>&1 >/dev/null | perl -lne 'print $1 if m{calling init: (/.*vgpreload_.*)}')
-    dracut_install $_valgrind_libs
+    valgrind_bins="$(strace -e execve valgrind /bin/true 2>&1 >/dev/null | perl -lne 'print $1 if /^execve\("([^"]+)"/')"
+    dracut_install "$valgrind_bins"
```

Comment:
@mrc0mmand I know it's been a year since `valgrind` broke but now that `dfuzzer` is on the way it would be great if it was possible to install it. I wonder what would be the right fix here in terms of making `shellcheck` happy. Should it be replaced with
```diff
-    valgrind_bins="$(strace -e execve valgrind /bin/true 2>&1 >/dev/null | perl -lne 'print $1 if /^execve\("([^"]+)"/')"
-    image_install "$valgrind_bins"
+    readarray -t valgrind_bins < <(strace -e execve valgrind /bin/true 2>&1 >/dev/null | perl -lne 'print $1 if /^execve\("([^"]+)"/')
+    image_install "${valgrind_bins[@]}"
```
or should the warning be suppressed?

Reply:
Oh, crap, I didn't mean to break it in the first place.

I'd prefer the `readarray` approach if it's not a big hassle (at least a quick test shows that it should work fine). If it would complicate things too much, suppressing it is definitely fine as well.

Reply:
I opened https://github.com/systemd/systemd/pull/23257

Reply:
To be fair `valgrind` was (and still is) kind of broken at runtime due `strip_binaries` (which has to be commented out) before this PR was opened so I think it's fair to say that this PR didn't break it :-) It would probably be nice to integrate it better into the testsuite but running PID1 under Valgrind can't be automated at this point unfortunately.